Crash when try to install uninstalled stickers pack

**1. Crash:**

Steps to reproduce:
- buy a new stickers
- uninstall it
- try to install it again
- app crash

<img width="428" alt="2" src="https://github.com/status-im/status-desktop/assets/86303051/ecdf3ddb-7de7-49fe-bf3b-26ff84cfa74c">


**2. Issue:**

Steps to reproduce:
- buy a new stickers
- use some stickers from the bought stickers pack
- uninstall stickers
- there are uninstalled stickers in the recently used stickers section, but there should not be any sticker that doesn't belong to the installed stickers displayed in that section

<img width="416" alt="3" src="https://github.com/status-im/status-desktop/assets/86303051/1503822e-f23e-480c-8cd2-49eef864e5c4">

Ok, I was able to reproduce once. It seems very random, because it worked all the other times I tried.
The stack trace isn't that helpful either:
```
DBG 2023-11-09 14:42:59.115-05:00 NewBE_callPrivateRPC                       topics="rpc" tid=408258 file=core.nim:27 rpc_method=stickers_uninstall
DBG 2023-11-09 14:42:59.118-05:00 NewBE_callPrivateRPC                       topics="rpc" tid=408258 file=core.nim:27 rpc_method=stickers_installed
DBG 2023-11-09 14:42:59.120-05:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=408276 file=threadpool.nim:54 messageType=AsyncGetRecentStickersTaskArg:ObjectType threadid=408276 task="{\"$type\":\"AsyncGetRecentStickersTaskArg:ObjectType\",\"vptr\":94046774418944,\"slot\":\"onAsyncGetRecentStickersDone\",\"tptr\":94046718839904}"
DBG 2023-11-09 14:42:59.120-05:00 NewBE_callPrivateRPC                       topics="rpc" tid=408276 file=core.nim:27 rpc_method=stickers_recent
DBG 2023-11-09 14:42:59.974-05:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=408275 file=threadpool.nim:54 messageType=InstallStickerPackTaskArg:ObjectType threadid=408275 task="{\"$type\":\"InstallStickerPackTaskArg:ObjectType\",\"packId\":\"0\",\"chainId\":1,\"hasKey\":false,\"vptr\":94046774418944,\"slot\":\"onStickerPackInstalled\",\"tptr\":94046718815248}"
INF 2023-11-09 14:42:59.975-05:00 shutting down...                           topics="status-app" tid=408258 file=nim_status_client.nim:173
INF 2023-11-09 14:42:59.975-05:00 logging out...                             topics="app-controller" tid=408258 file=app_controller.nim:287
Traceback (most recent call last)
/home/jonathan/dev/status-desktop/src/nim_status_client.nim(208) nim_status_client
/home/jonathan/dev/status-desktop/src/nim_status_client.nim(181) mainProc
/home/jonathan/dev/status-desktop/src/app/boot/app_controller.nim(324) delete
/home/jonathan/dev/status-desktop/src/app_service/service/stickers/service.nim(96) delete
/home/jonathan/dev/status-desktop/vendor/nimqml/src/nimqml/private/qobject.nim(67) delete
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
make: *** [Makefile:866: run-linux] Segmentation fault (core dumped)
```

--- a/src/app/modules/main/stickers/models/sticker_pack_list.nim
+++ b/src/app/modules/main/stickers/models/sticker_pack_list.nim
@@ -122,8 +122,11 @@ QtObject:
     self.packs.apply(proc(it: var StickerPackView) =
       if it.pack.id == packId:
         it.installed = installed
+        if installed:
+          it.bought = true
         it.pending = pending)
-    self.dataChanged(index, index, @[StickerPackRoles.Installed.int, StickerPackRoles.Pending.int])
+    self.dataChanged(index, index, @[StickerPackRoles.Installed.int, StickerPackRoles.Bought.int,
+      StickerPackRoles.Pending.int])
 
   proc getStickers*(self: StickerPackList): QVariant {.slot.} =
     let packInfo = self.packs[self.packIdToRetrieve]

--- a/src/app/modules/main/stickers/module.nim
+++ b/src/app/modules/main/stickers/module.nim
@@ -120,7 +120,8 @@ method onUserAuthenticated*(self: Module, password: string) =
       self.tmpBuyStickersTransactionDetails.eip1559Enabled
     )
     if response.error.isEmptyOrWhitespace():
-      self.view.stickerPacks.updateStickerPackInList(self.tmpBuyStickersTransactionDetails.packId, false, true)
+      self.view.stickerPacks.updateStickerPackInList(self.tmpBuyStickersTransactionDetails.packId, installed = false,
+        pending = true)
     self.view.transactionWasSent(chainId = response.chainId, txHash = response.txHash, error = response.error)
 
 method obtainMarketStickerPacks*(self: Module) =
@@ -250,10 +251,10 @@ method getChainIdForStickers*(self: Module): int =
   return self.controller.getChainIdForStickers()
 
 method stickerTransactionConfirmed*(self: Module, trxType: string, packID: string, transactionHash: string) =
-  self.view.stickerPacks.updateStickerPackInList(packID, true, false)
+  self.view.stickerPacks.updateStickerPackInList(packID, installed = true, pending = false)
   self.controller.installStickerPack(packID)
   self.view.emitTransactionCompletedSignal(true, transactionHash, packID, trxType)
 
 method stickerTransactionReverted*(self: Module, trxType: string, packID: string, transactionHash: string) =
-  self.view.stickerPacks.updateStickerPackInList(packID, false, false)
+  self.view.stickerPacks.updateStickerPackInList(packID, installed = false, pending = false)
   self.view.emitTransactionCompletedSignal(false, transactionHash, packID, trxType)

--- a/src/app_service/service/stickers/async_tasks.nim
+++ b/src/app_service/service/stickers/async_tasks.nim
@@ -7,12 +7,13 @@ type
     packId*: string
     fromAddress*: string
     uuid*: string
+  
   ObtainMarketStickerPacksTaskArg = ref object of QObjectTaskArg
     chainId*: int
   InstallStickerPackTaskArg = ref object of QObjectTaskArg
     packId*: string
     chainId*: int
-    hasKey*: bool
+  
   AsyncGetRecentStickersTaskArg* = ref object of QObjectTaskArg
   AsyncGetInstalledStickerPacksTaskArg* = ref object of QObjectTaskArg
 
@@ -68,9 +69,7 @@ const obtainMarketStickerPacksTask: Task = proc(argEncoded: string) {.gcsafe, ni
 
 const installStickerPackTask: Task = proc(argEncoded: string) {.gcsafe, nimcall.} =
   let arg = decode[InstallStickerPackTaskArg](argEncoded)
-  if not arg.hasKey:
-    arg.finish(false)
-    return
+
   var installed = false
   try:
     let installResponse = status_stickers.install(arg.chainId, arg.packId)

--- a/src/app_service/service/stickers/service.nim
+++ b/src/app_service/service/stickers/service.nim
@@ -174,10 +174,6 @@ QtObject:
       error "Error reverting sticker transaction", message = getCurrentExceptionMsg()
 
   proc init*(self: Service) =
-    # TODO redo the connect check when the network is refactored
-    # if self.status.network.isConnected:
-    self.obtainMarketStickerPacks() # TODO: rename this to obtain sticker market items
-
     self.events.on(PendingTransactionTypeDto.BuyStickerPack.event) do(e: Args):
       var receivedData = TransactionMinedArgs(e)
       if receivedData.success:
@@ -258,16 +254,16 @@ QtObject:
 
     for stickerPack in availableStickers:
       if self.marketStickerPacks.contains(stickerPack.id): continue
-      let isBought = stickerPack.status == StickerPackStatus.Purchased
+
       self.marketStickerPacks[stickerPack.id] = stickerPack
       self.events.emit(SIGNAL_STICKER_PACK_LOADED, StickerPackLoadedArgs(
         stickerPack: stickerPack,
         isInstalled: false,
-        isBought: isBought,
+        isBought: stickerPack.status == StickerPackStatus.Purchased,
         isPending: false
       ))
 
-    let chainId = self.networkService.getNetworkForStickers().chainId
+    # TODO move this to be async
     let pendingStickerPacksResponse = status_stickers.pending() 
     for (packID, stickerPackJson) in pendingStickerPacksResponse.result.pairs():
         if self.marketStickerPacks.contains(packID): continue
@@ -379,7 +375,6 @@ QtObject:
         error "error loading installed sticker packs", msg = error.message
         return
 
-      var stickerPacks: Table[string, StickerPackDto] = initTable[string, StickerPackDto]()
       for (packID, stickerPackJson) in rpcResponseObj{"result"}.pairs():
         self.installedStickerPacks[packID] = stickerPackJson.toStickerPackDto()
       self.events.emit(SIGNAL_LOAD_INSTALLED_STICKER_PACKS_DONE, StickerPacksArgs(packs: self.installedStickerPacks))
@@ -403,20 +398,23 @@ QtObject:
       slot: "onStickerPackInstalled",
       chainId: self.networkService.getNetworkForStickers().chainId,
       packId: packId,
-      hasKey: self.marketStickerPacks.hasKey(packId)
     )
     self.threadpool.start(arg)
 
   proc onStickerPackInstalled*(self: Service, installedPackJson: string) {.slot.} =
     let installedPack = Json.decode(installedPackJson, tuple[packId: string, installed: bool])
     if installedPack.installed:
+      if self.marketStickerPacks.hasKey(installedPack.packId):
+        self.marketStickerPacks[installedPack.packId].status = StickerPackStatus.Installed
       self.events.emit(SIGNAL_STICKER_PACK_INSTALLED, StickerPackInstalledArgs(
         packId: installedPack.packId
       ))
+    else:
+      error "Sticker pack did not get installed", packId = installedPack.packId
     
   proc uninstallStickerPack*(self: Service, packId: string) =
     try:
-      let installedResponse = status_stickers.uninstall(packId)
+      discard status_stickers.uninstall(packId)
     except RpcException:
       error "Error removing installed sticker", message = getCurrentExceptionMsg()
 

